src/quacc/atoms/phonons.py

```diff
@@ -62,24 +66,38 @@ def get_phonopy(
     """
     phonopy_kwargs = phonopy_kwargs or {}
 
-    structure = AseAtomsAdaptor().get_structure(atoms)
+    fixed_atoms = fixed_atoms or []
+    fixed_atoms = np.array([i in fixed_atoms for i in range(len(atoms))])
+
+    structure = AseAtomsAdaptor.get_structure(atoms)
     structure = SpacegroupAnalyzer(
         structure, symprec=symprec
     ).get_symmetrized_structure()
+    atoms = AseAtomsAdaptor.get_atoms(structure)
+
+    fixed, non_fixed = atoms[fixed_atoms], atoms[~fixed_atoms]
+
+    non_fixed = AseAtomsAdaptor.get_structure(non_fixed)
 
     if supercell_matrix is None and min_lengths is not None:
         n_supercells = np.round(np.ceil(min_lengths / atoms.cell.lengths()))
-        supercell_matrix = np.diag([n_supercells, n_supercells, n_supercells])
+        supercell_matrix = np.diag(np.diag([n_supercells, n_supercells, n_supercells]))
 
-    phonopy_atoms = get_phonopy_structure(structure)
+    phonopy_atoms = get_phonopy_structure(non_fixed)
     phonon = phonopy.Phonopy(
         phonopy_atoms,
         symprec=symprec,
         supercell_matrix=supercell_matrix,
         **phonopy_kwargs,
     )
     phonon.generate_displacements(distance=displacement)
-    return phonon
+
+    if len(fixed) > 0:
+        fixed = get_phonopy_structure(AseAtomsAdaptor.get_structure(fixed))
+        fixed = phonopy.structure.cells.get_supercell(fixed, supercell_matrix)
+        fixed = AseAtomsAdaptor.get_atoms(get_pmg_structure(fixed))
```

Comment:
These are three different types of objects yet they all have the same variable name; a bit confusing. This should be clarified.